### PR TITLE
refactor: rename `with_error_fn` into `when`

### LINKED ISSUE #11: Pick a more appropriate name for `with_error_fn`?
@Xuanwo This crate is awesome! I never thought of a retry mechanism that could be implemented in such an elegant way. :+1:

However the function name `with_error_fn` is somewhat ambiguous for me. I cannot guess what it is used for without reading the comments.

How about renaming it to `continue_if` or `break_if` (on the contrary), following the naming convention in [`std::ops::ControlFlow`](https://github.com/rust-lang/rust/blob/08871139915b95ec868aff807f224f78d00f4311/library/core/src/iter/traits/iterator.rs#L2166-L2172):

```rust
let content = fetch
    .retry(ExponentialBackoff::default())
    .continue_if(|e| e.to_string() != "EOF") // continue if error is not "EOF"
    .await?;
```

v.s.

```rust
let content = fetch
    .retry(ExponentialBackoff::default())
    .break_if(|e| e.to_string() == "EOF") // stop retrying when error is "EOF"
    .await?;
```

Personally I think `break_if` is a good choice. What do you think of it?

### PR DESCRIPTION
**Pull request by wfxr**
Rename `with_error_fn` into `when`.

Closes #11

### CODE REVIEW

File: src/retry.rs [hunk 1]
```diff
@@ -152,8 +152,37 @@ where
...
     /// }
     /// ```
+    #[deprecated(since = "0.0.3", note = "please use `Retry::when` instead")]
     pub fn with_error_fn(mut self, error_fn: fn(&E) -> bool) -> Self {
-        self.error_fn = error_fn;
+        self.retryable_fn = error_fn;
+        self
+    }
+
+    /// Set `retryable_fn` of retry
```
Comment (Xuanwo) [inline src/retry.rs:161]: Please don't leak our internal struct name.

Instead, we can:

```rust
/// Set the conditions for retrying.
///
/// If not specifed, we treat all errors as retryable.
```

File: src/retry.rs [hunk 2]
```diff
@@ -85,7 +85,7 @@ pub struct Retry<
     FutureFn: FnMut() -> Fut,
 > {
     backoff: B,
-    error_fn: fn(&E) -> bool,
+    retryable_fn: fn(&E) -> bool,
```
Comment (Xuanwo) [inline src/retry.rs:88]: Maybe using `retryable` here is enough?

Comment (wfxr) [reply src/retry.rs:88]: 👍 I also prefer `retryable`.

**Review by Xuanwo — APPROVED** [review]
Great!

**Pull request merged**